src/runtime/pymodule.cs

```diff
@@ -37,5 +40,40 @@ public static PyModule FromString(string name, string code)
             PythonException.ThrowIfIsNull(m);
             return new PyModule(ref m);
         }
+
+        public static PyModule Create(string name, string filename = "none")
+        {
+            NewReference op;
+
+            // first check if there is an existing module
+            BorrowedReference modules = Runtime.PyImport_GetModuleDict();
+            BorrowedReference m = Runtime.PyDict_GetItemString(modules, name);
+            if(!m.IsNull && Runtime.PyObject_TypeCheck(m, new BorrowedReference(Runtime.PyModuleType)))
+            {
+                // there is already a module by this name
+                op = new NewReference(m);
+                return new PyModule(ref op);
+            }
+
+            // create a new module
+            op = Runtime.PyModule_New(name);
+            PythonException.ThrowIfIsNull(op);
+
+            // setup the module basics (__builtins__ and __file__)
+            BorrowedReference globals = Runtime.PyModule_GetDict(new BorrowedReference(op.DangerousGetAddress()));
+            PythonException.ThrowIfIsNull(globals);
+            BorrowedReference __builtins__ = Runtime.PyEval_GetBuiltins();
+            PythonException.ThrowIfIsNull(__builtins__);
+            int rc = Runtime.PyDict_SetItemString(globals, "__builtins__", __builtins__);
+            PythonException.ThrowIfIsNotZero(rc);
+            rc = Runtime.PyDict_SetItemString(globals, "__file__", new BorrowedReference(filename.ToPython().Handle));
```

Comment:
The default value for `filename` should be `null` and when it is null `__file__` should simply not be set.